### PR TITLE
ripcord: init at 0.4.18

### PR DESCRIPTION
**Pull request by MDeltaX**
###### Motivation for this change

> Ripcord is a desktop chat client for group-centric services like Slack and Discord. It provides a traditional compact desktop interface designed for power users. It's not built on top of web browser technology: it responds quickly to input, sips gently from computer resources, and gets out of your way. It does voice chat, too.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ldesgoui @MP2E @tadeokondrak

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/Ripcord.desktop $out/share/applications/Ripcord.desktop
+    install -m 444 -D ${appimageContents}/Ripcord_Icon.png $out/share/icons/hicolor/512x512/apps/Ripcord_Icon.png
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A desktop chat client for Discord and Slack";
+    homepage = "https://cancel.fm/ripcord/";
```
Comment (tadeokondrak) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix]: Is it really redistributable?

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "19nv5pq8643l57hr5bdgn84rrw3ak2p9s16gr202nvi62flnl44p";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in appimageTools.wrapType2 rec {
+  inherit name src;
+
+  extraInstallCommands = ''
```
Comment (tadeokondrak) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix]: I think we should wrap the binary to set `RIPCORD_ALLOW_UPDATES=0` so it doesn't try to automatically update itself and fail

Comment (HeapUnderfl0w) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: afaik ripcord never attempts to self update. Ripcord only queries the update URL to show update notifications in Ripcord itself (which the `RIPCORD_ALLOW_UPDATES` disables).

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 1 above)
Comment (MicrosoftTakeover) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: The author has stated in Discord that it's "all rights reserved but redistribution without modifications is allowed (shareware)"
https://discordapp.com/channels/231664689694638081/231664689694638081/467792133319950337

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 2 above)
Comment (MicrosoftTakeover) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: I personally would not like update notifications for software delivered via package manger. You end up in annoying situations where there's a new version released but it's not packaged yet so you get notified over and over.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -4101,6 +4101,12 @@
     githubId = 1377571;
     name = "Matthew S. Daiter";
   };
+  MDeltaX = {
```
Comment (alexarice) [inline maintainers/maintainer-list.nix]: The addition to `maintainer-list.nix` should be in a separate commit named `maintainers: add MDeltaX`

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchurl {
+    url = "https://cancel.fm/dl/${pname}-${version}-x86_64.AppImage";
+    sha256 = "19nv5pq8643l57hr5bdgn84rrw3ak2p9s16gr202nvi62flnl44p";
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in appimageTools.wrapType2 rec {
+  inherit name src;
```
Comment (alexarice) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix]: Does this not accept `pname`?

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl, appimageTools }:
+
+let
+  pname = "Ripcord";
+  version = "0.4.18";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "https://cancel.fm/dl/${pname}-${version}-x86_64.AppImage";
```
Comment (alexarice) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix]:
```suggestion
    url = "https://cancel.fm/dl/${name}-x86_64.AppImage";
```
Since you've defined name anyway

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+in appimageTools.wrapType2 rec {
+  inherit name src;
+
+  extraInstallCommands = ''
+    mv $out/bin/${name} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/Ripcord.desktop $out/share/applications/Ripcord.desktop
+    install -m 444 -D ${appimageContents}/Ripcord_Icon.png $out/share/icons/hicolor/512x512/apps/Ripcord_Icon.png
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (alexarice) [inline pkgs/applications/networking/instant-messengers/ripcord/default.nix]: As far as I can see you do not use `stdenv` anywhere else so you could just import `lib` at the top instead

```suggestion
  meta = with lib; {
```

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 1 above)
Comment (MDeltaX) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: yep

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 4 above)
Comment (MDeltaX) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/appimage/default.nix#L56
apparently nope

File: pkgs/applications/networking/instant-messengers/ripcord/default.nix (hunk 1 above)
Comment (MicrosoftTakeover) [reply pkgs/applications/networking/instant-messengers/ripcord/default.nix]: https://cancel.fm/ripcord/shareware-redistribution/

**Comment by MDeltaX** [comment]
will come back to this in the next days

**Comment by Profpatsch** [comment]
Is this ready for merge?

**Comment by colemickens** [comment]
Hey @MDeltaX, do you think you'll get to this soon, and if not, do you mind if someone else pushes it along? Thanks for doing this work!

**Comment by colemickens** [comment]
Does this need some Qt wrap hooks? I'm not able to run this on Wayland due to the usual Qt platform plugin err.

**Comment by bqv** [comment]
Christ this is old. Sorry this nobody ever bothered to merge this. My version is more up to date at this point anyway

**Comment by Profpatsch** [comment]
Then let’s close in favor of the other PR.
